src/prop-converter.js

```diff
@@ -41,16 +45,18 @@ function propConverter(ComposedComponent) {
       // child is passed, as well as if the child is falsy.
       workingProps.children = React.Children.toArray(props.children);
 
-      // FlipMove does not support stateless functional components.
-      // Check to see if any supplied components won't work.
-      // If the child doesn't have a key, it means we aren't animating it.
-      // It's allowed to be an SFC, since we ignore it.
-      const noStateless = workingProps.children.every(child =>
-         !isElementAnSFC(child) || typeof child.key === 'undefined'
-      );
-
-      if (!noStateless) {
-        console.warn(statelessFunctionalComponentSupplied());
+      if (process.env.NODE_ENV !== 'production') {
+        // FlipMove does not support stateless functional components.
+        // Check to see if any supplied components won't work.
+        // If the child doesn't have a key, it means we aren't animating it.
+        // It's allowed to be an SFC, since we ignore it.
+        const noStateless = workingProps.children.every(child =>
+           !isElementAnSFC(child) || typeof child.key === 'undefined'
+        );
+
+        if (!noStateless) {
+          console.warn(statelessFunctionalComponentSupplied());
+        }
```

Comment:
Yeah, I just have a natural aversion to nested `if` statements, but you're probably right in that it's simpler to understand inline.

I may revise it later, but I'm happy to accept it as-is. I can tinker with it and see if it makes more sense.